File: 2023-2024/sourcecode/mod3/RetrofitBasicsJava/app/src/main/java/com/example/retrofitbasicsjava/ui/PostAdapter.java
package com.example.retrofitbasicsjava.ui;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.recyclerview.widget.ListAdapter;

import com.example.retrofitbasicsjava.data.Post;
import com.example.retrofitbasicsjava.databinding.RvItemBinding;

public class PostAdapter extends ListAdapter<Post, PostViewHolder> {
    public PostAdapter(PostComparator postComparator) {
        super(postComparator);
    }

    @Override
    public PostViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        RvItemBinding binding = RvItemBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false
        );
        return new PostViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(PostViewHolder holder, int position) {
        Post item = getItem(position);
        holder.bind(item);
    }
}

File: 2023-2024/sourcecode/mod4/lab12/RetrofitBasicsKotlin/app/src/main/java/com/example/retrofitbasicskotlin/ui/PostAdapter.kt
package com.example.retrofitbasicskotlin.ui

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import com.example.retrofitbasicskotlin.data.Post
import com.example.retrofitbasicskotlin.databinding.RvItemBinding

class PostAdapter(userComparator: PostComparator) : ListAdapter<Post, PostViewHolder>(userComparator) {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PostViewHolder {
        return PostViewHolder(
            RvItemBinding.inflate(
                LayoutInflater.from(parent.context), parent, false
            )
        )
    }

    override fun onBindViewHolder(holder: PostViewHolder, position: Int) {
        val item = getItem(position)
        holder.bind(item)
    }
}